File: hrapp/eg1.py
import cx_Oracle as oracle;
try:
	connection=oracle.connect("hr/hr@localhost:1521/xepdb1")
	print("Connection to Oracle database")
	cursor=connection.cursor()
	cursor.execute("select * from employees")
	for row in cursor:
		print(row)
	cursor.close()
except oracle.DatabaseError as e:
		print(f"Problem {e}");
 finally:
	connection.close()
	print("Connection to Oracle has been closed")

File: hrapp/eg1.js
const oracle=require('oracledb');
async function testConnection()
{
var connection=null;
try
{
conection=await oracle.getConnection({
"user": "hr",
"password": "hr",
"connectionString": "localhost:1521/xepdb1"
});
console.log("Connection to oracle");
resultSet=await connection.execute("select * from employees");
resultSet.rows.forEach((row)=>{
console.log(row);
});
console.log(`Number of employees ${resultSet.rows.length}`);
}catch(err)
{
console.log(err.message);
}
finally
{
if(connection)
{
try
{
await connection.close();
console.log("Connection to oracle has been closed");
}catch(err)
{
console.log(err);
}
}
}
}

testConnection();